weasyl/profile.py

```diff
@@ -285,19 +285,22 @@ def select_relation(userid, otherid):
             "is_self": userid == otherid,
         }
 
-    query = d.execute("""
+    query = """
         SELECT
-            (SELECT EXISTS (SELECT 0 FROM watchuser WHERE (userid, otherid) = (%i, %i))),
-            (SELECT EXISTS (SELECT 0 FROM frienduser WHERE userid IN (%i, %i) AND otherid IN (%i, %i) AND settings !~ 'p')),
-            (SELECT EXISTS (SELECT 0 FROM ignoreuser WHERE (userid, otherid) = (%i, %i))),
-            (SELECT EXISTS (SELECT 0 FROM frienduser WHERE (userid, otherid) = (%i, %i) AND settings ~ 'p'))
-    """, [userid, otherid, userid, otherid, userid, otherid, userid, otherid, userid, otherid], ["single"])
+            (SELECT EXISTS (SELECT 0 FROM watchuser WHERE (userid, otherid) = ({user_id}, {other_id}))),
+            (SELECT EXISTS (SELECT 0 FROM frienduser WHERE userid IN ({user_id}, {other_id}) AND otherid IN ({user_id}, {other_id}) AND settings !~ 'p')),
+            (SELECT EXISTS (SELECT 0 FROM ignoreuser WHERE (userid, otherid) = ({user_id}, {other_id}))),
+            (SELECT EXISTS (SELECT 0 FROM frienduser WHERE (userid, otherid) = ({user_id}, {other_id}) AND settings ~ 'p')),
+            (SELECT EXISTS (SELECT 0 FROM watchuser WHERE (userid, otherid) = ({other_id}, {user_id})))
+    """.format(user_id=userid, other_id=otherid)
+    query = d.execute(query, None, ["single"])
```

Comment:
It still works with the following code, successfully getting `"ignore"` status from `result[2]`, right?